_posts/en/newsletters/2024-08-09-newsletter.md

```diff
@@ -0,0 +1,342 @@
+  or to require pool miners use a public identity so that bad actors can
+  be banned.
+
+  One solution Towns proposes is for pools to provide multiple block
+  templates, allowing each miner to choose their preferred template.
+  This is similar to the existing system used by [Ocean Pool][].  Shares
+  submitted based on a pool-created templates can be validated quickly
+  and with a minimum amount of bandwidth.  This prevents the invalid
+  shares attack that pays 100% but does not help with the approximately
+  99.9% profitable block withholding attack.
+
+  To address the block withholding attack, Towns updates an idea
+  [originally proposed][rosenfeld pool] by Meni Rosenfeld in 2011 that
+  describes how a conceptually simple hard fork could prevent a pool
+  miner from knowing whether any particular share had enough PoW to be a
+  valid block, making them _oblivious shares_.  An attacker who can't
+  discriminate between valid-block PoW and share PoW can only deprive a
+  pool of valid-block income at the same rate the attacker deprives
+  themselves of share income.  The approach has downsides:
+
+  - *SPV-visible hard fork:* all hard fork proposals require all full
+    nodes to upgrade.  However, many proposals (such as simple block
+    size increase proposals like [BIP103][]) do not require upgrades of
+    lightweight clients that use _simplified payment validation_ (SPV).
+    This proposal changes how the block header field is interpreted and
+    so would require all lightweight clients to also upgrade.  Towns
+    does offer an alternative that wouldn't necessarily require light
+    clients to upgrade, but it would reduce their security
+    significantly.
+
+  - *Requires pool miners use a private template from the pool:* not
+    only would a template be required to prevent the 100% invalid shares
+    attack but the pool would need to keep that template secret from
+    pool miners until after all shares generated using that template
+    were received.  The pool could use that to trick miners into
+    producing PoW for transactions the miners would object to.  However,
+    expired templates could be published to allow auditing.  Most modern
+    pools generate a new template every few seconds, so any auditing
+    could be performed in near real-time, preventing a malicious pool
+    from tricking its miners for more than a few seconds.
+
+  - *Requires share submission:* one of the advantages of Stratum v2 (in
+    certain operating modes) is that an honest pool miner who finds a
+    block for the pool can immediately broadcast it to the Bitcoin P2P
+    network, allowing it to begin propagating even before the
+    corresponding share has reached the pool server.  With oblivious
+    shares, the share would need to be received by the pool and
+    converted into a full block before it could be broadcast.
+
+  Towns concludes by describing two of the motivations for fixing the block
```

Comment:
```suggestion
  Towns concludes by describing two motivations for fixing the block
```

Reply:
Leaving as-is.  I think "two of the motivations" expresses that there are other motivations; "two motivations" sounds like there's only two motivations.

Reply:
Well, then it would be "the two motivations".

Reply:
Still leaving as-is, for the reason stated.